cli/crates/backend/src/api/deploy.rs

```diff
@@ -41,42 +40,42 @@ pub async fn deploy() -> Result<(), ApiError> {
     let project_metadata: ProjectMetadata =
         serde_json::from_str(&project_metadata_file).map_err(|_| ApiError::CorruptProjectMetadataFile)?;
 
-    let (tar_file, tar_file_path) = tempfile::NamedTempFile::new()
-        .map_err(ApiError::CreateTempFile)?
-        .into_parts();
+    let tar_file_path = tokio::task::spawn_blocking(|| {
+        let (tar_file, tar_file_path) = tempfile::NamedTempFile::new()
+            .map_err(ApiError::CreateTempFile)?
+            .into_parts();
 
-    let tar_file: tokio::fs::File = tar_file.into();
-    let tar_file = tar_file.compat();
+        let mut tar = tar::Builder::new(tar_file);
+        tar.mode(tar::HeaderMode::Deterministic);
```

Comment:
Just so I'm not missing anything, this is the fix itself right? The other change is `spawn_blocking` which I'm assuming isn't related to the specific issue

Reply:
Or actually it looks like we also switched to sync tar and that's the reason for blocking, ok